test/compilable/riia_ctor.d

```diff
@@ -11,23 +11,6 @@ class C
     this() nothrow {}
 }
 
-// https://issues.dlang.org/show_bug.cgi?id=17505
-struct Array
-{
-    int[] _payload;
-    ~this()
-    {
-        import core.stdc.stdlib : free;
-        free(_payload.ptr);
-    }
-}
-
-class Scanner
-{
-    Array arr;
-    this() @safe {}
-}
```

Comment:
Move into fail_compilation?

Reply:
Reverted this and (ab)used it to test `-revert=dtorfields` ... which fails ATM.

Reply:
`parseCLIOption` doesn't handle reverts correctly. This is the generated code:

```
case "all":
        params.noDIP25 = true;
        params.markdown = true;
        params.dtorFields = true;
        return true;
case "dip25":
        params.noDIP25 = true;
        return true;

case "markdown":
        params.markdown = true;
        return true;

case "dtorfields":
        params.dtorFields = true;
        return true;
```

`-revert=dip25` only works because of the second option `noDIP25`. I think a better approach would be to  store  revertable switches as a three state boolean (e.g `set`, `default`, `unset`)